tests/cms/views/interactions/test_vote_action.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+from django.contrib.messages.storage.fallback import FallbackStorage
+from django.test import RequestFactory, TestCase
+
+from integreat_compass.cms.models import User, Vote
+from integreat_compass.cms.views.interactions import add_vote
+
+
+def get_request():
+    request = RequestFactory().post("", {"approval": "False"})
+    request.user = User.objects.get(pk=1)
+    setattr(request, "session", "session")
+    messages = FallbackStorage(request)
+    setattr(request, "_messages", messages)
+    return request
```

Comment:
That's probably more of a philosophical discussion, but in my opinion, mocking request objects is not sufficient for testing, because it does not realistically simulate how the user is using the system. Specific settings and middlewares could have side effects which are not covered by these tests, also it does not check whether the URL resolver is configured correctly, it ignores all potential errors that might occur during template rendering, etc...

Testing the specific methods of view classes does provide the additional value that it helps debugging a problem if a test fails (because of the limitations mentioned above, we can e.g. narrow down if it's a bug of the view function itself, or something else), but in my opinion these type of tests help most if there are existing view tests that provide a more realistic scenario.

So in the future, I think it would make sense to start with writing tests using the [test client](https://docs.djangoproject.com/en/4.2/topics/testing/tools/#the-test-client) and verifying whether making actual requests has the effect we expect. And only if time is left, we can improve these tests and write more fine-graded tests which help to identify the actual problem when a tests fails.